src/components/pages/MyProfilePage/ModalEditInterests/CurrentModalEdit/CurrentModalEdit.tsx

```diff
@@ -0,0 +1,111 @@
+import { Button, Input, ModalBox } from './../../../../common';
+import { ArrowBack } from '../../../ProfilePage/components/ArrowBack';
+import s from './CurrentModalEdit.module.css';
+import { DataProfile } from '../../../../../mocks/dataProfile';
+import { useAppSelector } from 'hooks/hooks';
+import { useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { updateDataProfile } from 'store/profileData/slice';
+import { CurrentModalEditProps } from 'components/pages/ProfilePage/types';
+export const CurrentModalEdit = ({
+  isOpen,
+  setOpen,
+  category,
+}: CurrentModalEditProps) => {
+  const dispatch = useDispatch();
+  const profileInfo = useAppSelector((state) => state.profileData);
+  const currentCategory = profileInfo.aboutProfile.find(
+    (it) => it.category === category,
+  );
+  const [count, setCount] = useState(0);
+  const [selectedCheckboxes, setSelectedCheckboxes] = useState<string[]>([]);
+  const handleCloseModal = () => {
+    setOpen(false);
+  };
+  const handleSave = () => {
+    dispatch(
+      updateDataProfile({
+        aboutProfile: profileInfo.aboutProfile.map((item) =>
+          item.category === category
+            ? { ...item, userSelection: selectedCheckboxes }
+            : item,
+        ),
+      }),
+    );
+    handleCloseModal();
+  };
+
+  const handleCheckboxChange = (option: string) => {
+    if (
+      currentCategory &&
+      selectedCheckboxes.length >= currentCategory?.maxCount &&
+      !selectedCheckboxes.includes(option)
+    ) {
+      return;
+    }
+
+    setSelectedCheckboxes((prevSelected) => {
+      const updatedSelection = selectedCheckboxes.includes(option)
+        ? prevSelected.filter((item) => item !== option)
+        : [...prevSelected, option];
+
+      setCount(updatedSelection.length);
+
+      return updatedSelection;
+    });
+  };
+
+  return (
+    <>
+      <ModalBox open={isOpen}>
+        <ArrowBack onClick={handleCloseModal} />
+        {currentCategory && (
+          <>
+            <div className={s.categoryEditHeader}>
+              <img src={currentCategory.img} alt={currentCategory.category} />
+              <p className={s.questionEdit}>{currentCategory.question}</p>
+
+              <p className={s.countEdit}>
+                {count}/{currentCategory.maxCount}
+              </p>
+            </div>
+            <div className={s.categoryValue}>
+              <ul className={s.categoryValueList}>
+                {currentCategory.value.map((option: string) => (
+                  <>
```

Comment:
VM45:1 Warning: Each child in a list should have a unique "key" prop.

```suggestion
                  <Fragment key={option}>
```